dump-event-schema-compiler-plugin/src/main/scala/org/virtuslab/ash/DumpEventSchemaCompilerPluginComponent.scala

```diff
@@ -41,8 +41,9 @@ class DumpEventSchemaCompilerPluginComponent(val options: DumpEventSchemaOptions
         def shouldIgnoreType(tpe: Type) = shouldIgnoreSymbol(tpe.typeSymbol)
 
         val foundUsedClasses: List[(Type, Position)] = body.collect {
-          case x: TypeTree if genericsNames.contains(x.tpe.dealias.typeSymbol.fullName) => (x.tpe.typeArgs.head, x.pos)
-        }
+          case x: TypeTree if genericsNames.contains(x.tpe.dealias.typeSymbol.fullName) =>
+            x.tpe.typeArgs.map((_, x.pos))
```

Comment:
Whoops... I can't match how this change (where nothing is mentioned about the state classes) corresponds to including states in the dump 🤔

Reply:
Awww crap okay I see now... but still, it's pretty unclear :///

Reply:
I'd definitely add some more clarity/named params/tuple decomposition here... lemme think...

Reply:
I compile doesn't like tuple decomposition with outside knowledge
```scala
#It would fail on the following inputs: List(_), Nil
val List(event, state) = x.tpe.typeArgs.takeRight(2)
```